cmd/conduit/root/root.go

```diff
@@ -0,0 +1,262 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package root
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/conduitio/conduit/cmd/conduit/internal"
+	"github.com/conduitio/conduit/cmd/conduit/root/pipelines"
+	"github.com/conduitio/conduit/pkg/conduit"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/ecdysis"
+)
+
+var (
+	_ ecdysis.CommandWithFlags       = (*RootCommand)(nil)
+	_ ecdysis.CommandWithExecute     = (*RootCommand)(nil)
+	_ ecdysis.CommandWithDocs        = (*RootCommand)(nil)
+	_ ecdysis.CommandWithSubCommands = (*RootCommand)(nil)
+)
+
+type RootFlags struct {
+	// Global flags -----------------------------------------------------------
+
+	// Conduit configuration file
+	ConduitConfigPath string `long:"config.path" usage:"global conduit configuration file" persistent:"true" default:"./conduit.yaml"`
+
+	// Version
+	Version bool `long:"version" short:"v" usage:"show current Conduit version" persistent:"true"`
+
+	// Root flags -------------------------------------------------------------
+
+	// Database configuration
+	DBType                     string `long:"db.type" usage:"database type; accepts badger,postgres,inmemory,sqlite"`
```

Comment:
WDYT about moving these tags to Conduit's `Config` struct? It looks like that would make the `Config` struct the single source of truth. It should then be possible to build the flags by just traversing the `Config` fields. Also, when building the config.yaml file in the `InitCommand`, we would have everything in the config struct (name, usage, and values).

Btw, even if we find that this is possible and useful, IMO it can be done in a different PR.

Reply:
https://github.com/ConduitIO/ecdysis/pull/1